flink_scrat/utils.py

```diff
@@ -30,3 +31,47 @@ def setup_logging():
         logging.config.dictConfig(config)
     else:
         logging.basicConfig(level=default_level)
+
+def handle_response(req_response):
+        req_response.raise_for_status()
+        return req_response.json()
+
+def find_manager_address(yarn_address, yarn_port, app_name):
+    apps_info = _get_apps_info(yarn_address, yarn_port)
+
+    running_app = _get_running_app(apps_info, app_name)
+
+    rpc_addres_info = _build_rpc_address_info(running_app)
+
+    return rpc_addres_info
+
+def _get_apps_info(yarn_address, yarn_port):
+    route = "http://{}:{}/ws/v1/cluster/apps".format(yarn_address, yarn_port)
+    
+    result = handle_response(requests.get(route))
+
+    apps_info = result['apps']['app']
+
+    return apps_info
+
+def _get_running_app(apps_info, app_name):
+    running_apps = list(filter(lambda x: x['state'] == 'RUNNING' and x['name'] == app_name, apps_info))
+
+    if len(running_apps) == 0:
+        raise Exception("No app found with state=<RUNNING> and name=<{}>".format(app_name))
+    elif len(running_apps) > 1:
+        raise Exception("More then one app found with state=<RUNNING> and name=<{}>".format(app_name))
```

Comment:
You should define you own exceptions for these cases

Reply:
👍